src/quicklist.c

```diff
@@ -374,24 +374,28 @@ REDIS_STATIC void __quicklistCompress(const quicklist *quicklist,
     if (!in_depth)
         quicklistCompressNode(node);
 
+
     /* At this point, forward and reverse are one node beyond depth */
     quicklistCompressNode(forward);
     quicklistCompressNode(reverse);
 }
 
 #define quicklistCompress(_ql, _node)                                          \
     do {                                                                       \
-        if ((_node)->recompress)                                               \
+        if ((_node)->recompress) {                                             \
+            assert(_node != _ql->head && _node != _ql->tail);                  \
```

Comment:
Should it be moved to `quicklistCompressNode`?

Reply:
in quicklistCompressNode, I don't have the quicklist. only a specific node.

Reply:
In that case, `assert(node->prev && node->next)` would be more appropriate.